Makefile

```diff
@@ -1,5 +1,31 @@
 # Try to locate the AWS SDK if not specified with AWS_SDK_PATH
 MACHINE := $(shell gcc -dumpmachine)
+# macOS-specific dependency detection
+ifeq ($(shell uname), Darwin)
+  # OpenSSL via Homebrew
+  OPENSSL_PREFIX := $(shell brew --prefix openssl@3 2>/dev/null)
+  ifneq ($(OPENSSL_PREFIX),)
+    OPENSSL_INC := -I$(OPENSSL_PREFIX)/include
+    OPENSSL_LIB := -L$(OPENSSL_PREFIX)/lib
```

Comment:
Can this append to LIBS rather than being a separate var? (Seems like it might just simplify the commands below)